sbt-scalajs-bundler/src/main/scala/scalajsbundler/util/JS.scala

```diff
@@ -79,17 +93,35 @@ object JS {
 
   def `new`(ctor: JS, args: JS*): JS = JS(New(ctor.tree, args.map(_.tree).to[List]))
 
-  def toJson(obj: JS): String = show(obj.tree, isStat = false)
-
-  def show(tree: Tree, isStat: Boolean = true): String = {
-    val writer = new java.io.StringWriter
-    val printer = new Printers.JSTreePrinter(writer)
-    printer.printTree(tree, isStat)
-    writer.toString
-  }
-
   private val identifierSeq = new AtomicInteger(0)
   private def freshIdentifier(): String =
     s"x${identifierSeq.getAndIncrement()}"
 
 }
+
+final class JSON(tree: Tree) extends JSLike(tree)
+
+object JSON {
+
+  implicit lazy val position: Position = Position.NoPosition
+
+  def apply(tree: Tree): JSON = new JSON(tree)
+
+  /** Array literal. */
+  def arr(elems: JSON*): JSON = JSON(ArrayConstr(elems.map(_.tree).to[List]))
+
+  /** Boolean literal */
+  def bool(value: Boolean): JSON = JSON(BooleanLiteral(value))
+
+  /** Object literal */
+  def obj(fields: (String, JSON)*): JSON =
```

Comment:
Do you think that would work to:
- remove duplicate definitions between `JS` and `JSON` smart constructors (e.g. `str`, `obj`, etc.) ;
- replace references to `JSON` and `JS` with `JSLike`, when needed.

For instance the type signature of `JSON.obj` would be:

~~~ scala
def obj(fields: (String, JSLike)*): JSON
~~~

The type signature of `JS.let` would be:

~~~ scala
def let(value: JSLike)(usage: JS => JSLike): JS
~~~

If you think that would be too much effort I’m happy to merge the PR in its current state (can you just update the scaladoc description to mention that the `dependencies` and `devDependencies` are automatically generated from the corresponding sbt settings?).